packages/web/src/pages/pay-and-earn-page/components/CashWallet.tsx

```diff
@@ -0,0 +1,165 @@
+import { useCallback } from 'react'
+
+import { useIsManagedAccount, useUSDCBalance } from '@audius/common/hooks'
+import { Name, Status } from '@audius/common/models'
+import {
+  WithdrawUSDCModalPages,
+  useWithdrawUSDCModal,
+  useAddFundsModal
+} from '@audius/common/store'
+import { USDC } from '@audius/fixed-decimal'
+import {
+  Button,
+  Flex,
+  IconInfo,
+  IconLogoCircle,
+  IconLogoCircleUSDC,
+  Paper,
+  Text,
+  TextLink
+} from '@audius/harmony'
+import BN from 'bn.js'
+
+import { useModalState } from 'common/hooks/useModalState'
+import { make, track } from 'services/analytics'
+
+const messages = {
+  usdc: 'USDC',
+  earn: 'Earn USDC by selling your music',
+  buyAndSell: 'Buy and sell music with USDC',
+  learnMore: 'Learn More',
+  withdraw: 'Withdraw',
+  addFunds: 'Add Cash',
+  salesSummary: 'Sales Summary',
+  withdrawalHistory: 'Withdrawal History',
+  cashBalance: 'Cash Balance',
+  payoutWallet: 'Payout Wallet',
+  builtInWallet: 'Built-In Wallet',
+  transactionHistory: 'Transaction History'
+}
+
+export const CashWallet = () => {
+  const isManagedAccount = useIsManagedAccount()
+  const { onOpen: openWithdrawUSDCModal } = useWithdrawUSDCModal()
+  const { onOpen: openAddFundsModal } = useAddFundsModal()
+  const { data: balance, status: balanceStatus } = useUSDCBalance()
+  const [, setPayoutWalletModalOpen] = useModalState('PayoutWallet')
+
+  // Calculate the balance in cents by flooring to 2 decimal places then multiplying by 100
+  const usdcValue = USDC(balance ?? new BN(0)).floor(2)
```

Comment:
is there a way we can avoid using `bn.js`? can we just pass 0 into `USDC`?

Reply:
i feel like fixed-decimal has utils to do all the formatting you need, but might need to ping @rickyrombo for deets.